### ISSUE TITLE
Feature request

### ISSUE THREAD

**Issue by Regomy** [Nort721/GodsEye#431]
Add a check for nps as in another popular anti-cheat. 

How could it be?

In the config will be a line "Use NPCs".

If the check will be enabled, then there will be a string containing "at what number of violations will appear nps".

You can also add a string along the lines of "how many times must the player hit the NPC to get it kicked".

You can also add a line containing "whether the NPC will be visible or invisible".
 
You can also add a line like this:
"Will the NPC fly randomly near the player or only behind his back"
"The speed of his movement (for example 3 types: teleportation, smooth, fast")
"At what distance will the NPC be located"
"If the number of violations does not increase, after how long the npc will disappear"
"Maximum number of NPCs on the whole server" (If more than the specified number of NPCs gets used, until the old NPCs of other players disappear, new ones won't appear for new players).

**Comment by Izibane** [comment]
npcs are relatively.. useless
look at matrix, you can hit the bot and get yourself banned
if not antibot nullifies this entirely.

**Comment by Regomy** [comment]
> npcs are relatively.. useless
> look at matrix, you can hit the bot and get yourself banned
> if not antibot nullifies this entirely.

Why do you think so? If you do not like nps, does not mean that others will not use them, please note that most cheats do not even have anti-bot, I'm not talking about private clients and about free. Nps is a good enough thing and can serve as a nice addition, even if it will not be complete.

**Comment by 0xntpower** [comment]
GodsEye had a bot checking system in the past and it got removed because it was useless and took too much effort to maintain...
All clients these days have antibot modules, the ones that don't have a visible antibot module are most likely checking for bots in the background, in general bots can be fully bypassed by checking some specific easy values and this bypass will work no matter how good the bot is,

Trust me on that one I did a full investigation into this, I was very into bot checks in the past and coded anti-bot modules in the past as well (I was a client developer before I became an anticheat developer)

not to mention that bot checks take a lot of performance, and let's say you're making the most realistic bot, and making a lot of systems to eliminate false positives with smart movement, basically making the best bot check possible would take way way more performance just to make a bot check that can be bypassed super easily and have been bypassed for years now,

While instead of putting so much effort and giving up so much performance, you could use math and statistical analysis checks that take way way way less performance and detect way more stuff

Basically here is a pro and cons comparison:
pros:
- might detect some extremely old outdated clients that very few people use more quickly than a math check

cons:
- very hard to develop and maintain on an anticheat that supports a lot of versions
- takes a LOT of performance and server resources while detecting less than way more lightweight math checks
- interrupts players experience (and if it doesn't then it would be even easier to bypass)
- very easy to false (and if it doesn't then it would be even easier to bypass)

As you can see, it just doesn't worth it, while bot checks might be cool they are just not efficient or worth it
